### PR TITLE
nilrt-runmode-initramfs: Remove initramfs recipe and references

### PR DESCRIPTION
**Pull request by billpittman**
Remove the initramfs recipe since it is no longer needed by nilrt
boot sequence and unnecessarily complicates the boot logic.

This change will delete the enable_initramfs variable from the grub
environment during the installation of the runmode system.  If that
variable is not present, the system will not attempt to boot from
an initramfs, any implementations that require an initramfs in runmode
will not work with this change.

Signed-off-by: Bill Pittman <bill.pittman@ni.com>
Natinst-AzDO-ID: 1868559

### Testing
Provisioned a cRIO, PXI-8861, Hyper-V VM and QEMU VM with the nilrt-recovery-media-x64.iso containing these changes and verified MAX could recognize and install to all of them.

### CODE REVIEW

**Review by amstewart — CHANGES_REQUESTED** [review]
Do you need to do anything with the [`nisetbootmode(.functions)`](https://github.com/ni/meta-nilrt/blob/nilrt/master/hardknott/recipes-ni/ni-utils/files/nisetbootmode.functions#L6) files?

**Comment by billpittman** [comment]
@amstewart Yup nice catch I forgot to search for bootflags.d references too.  Got it now and there appears to be no other references to bootflags.  Also tested the scripts on an installed runmode&safemode system.

File: recipes-ni/ni-systemformat/files/nisystemformat
Comment (chaitu236) [inline recipes-ni/ni-systemformat/files/nisystemformat:653]: So have we never enabled encryption of configfs, rootfs partitions? I do see some code related to that but wasn't sure if it was in use.

File: recipes-core/initrdscripts/files/ni_provisioning.safemode
Comment (chaitu236) [inline recipes-core/initrdscripts/files/ni_provisioning.safemode:325]: So going forward, a target provisioned with old safemode won't be able to boot new runmode on targets with TPM chip. I guess this is acceptable to us? Should we add checks somewhere to ensure new runmode isn't installed on old safemode?

File: recipes-ni/ni-systemformat/files/nisystemformat
Comment (billpittman) [reply recipes-ni/ni-systemformat/files/nisystemformat:653]: Right, pretty sure its been broken for a while now,  The upstream API changed so most of the plumbing we have won't work even if we enabled it.

File: recipes-core/initrdscripts/files/ni_provisioning.safemode
Comment (billpittman) [reply recipes-core/initrdscripts/files/ni_provisioning.safemode:325]: Sort of, old safemode will still boot on all targets, however if enable_initramfs is set to 1 (because the target has a TPM chip, or whatever), the new runmode won't boot since the initramfs will be "missing on purpose due to this change".  If the old safemode has enable_initramfs==0, then it will boot the old safemode/new runmode combo just fine since neither one requires the initramfs.

Does that answer your question?

**Review by shruthi-ravi — COMMENTED** [review]
Since there are changes to so many files, can you enumerate the repercussions of these changes in the commit message? i.e. what is possible and not once this change goes in? Chaitanya seems to have asked a few questions related to that.
And can you also add a link to the azdo WI or WI number to the PR?
The azdo WI has a lot of the details I am requesting, but I would prefer if some of it were in the commit message too.

**Review by amstewart — APPROVED** [review]

**Comment by amstewart** [comment]
I'll wait until everyone on the review list has given their ACK before merging.

File: recipes-core/initrdscripts/files/ni_provisioning.safemode
Comment (chaitu236) [reply recipes-core/initrdscripts/files/ni_provisioning.safemode:325]: I understand that thats the case. I'm wondering if/how we will enforce it - right now I believe MAX doesn't stop us from installing hardknott runmode on a sumo safemode. Should we create a story to research/implement this?

Comment (chaitu236) [reply recipes-core/initrdscripts/files/ni_provisioning.safemode:325]: PS: Have we broken safemode compatibility for new releases like this before? @billpittman, @jeminor

Comment (billpittman) [reply recipes-core/initrdscripts/files/ni_provisioning.safemode:325]: I don't believe we've broken compatibility necessarily.  In the case of a TPM enabled system, then yes, but we've dropped that capability, so its a compatibility break in the sense that we don't support it.  Safemode still installs runmode, even an older safemode.

Comment (billpittman) [reply recipes-core/initrdscripts/files/ni_provisioning.safemode:325]: @chaitu236 Pointed out a specific case where TPM capable targets would require a safemode upgrade in order to properly install.  Without the update, MAX would "successfully" provision the target, but when it boots into the newly installed runmode, it would fail to find the initramfs file and force the target into  safemode with a software installation error.

Do we have a way to "force" a safemode upgrade from MAX or inform the user that the "problem" is the old safemode?  We used to have a mechanism like that, but it was removed a very long time ago and I don't remember if we put in something else in its place.

If we don't have a way to inform the user that there's a safemode upgrade required, we may have to delay/abandon this change for the moment until we can handle TPM capable systems.

Comment (amstewart) [reply recipes-core/initrdscripts/files/ni_provisioning.safemode:325]: I think we do have a way for MAX to notify customers that a safemode upgrade is required. Could you ask Dan and co.

Comment (jeminor) [reply recipes-core/initrdscripts/files/ni_provisioning.safemode:325]: There is a mechanism and can require it be invoked if needed.  Specifically, we can require that a system image install require >= some specific safemode.

I'm fine with this, and we can invoke that mechanism when we need it.

**Review by jeminor — APPROVED** [review]

File: recipes-core/initrdscripts/files/ni_provisioning.safemode
Comment (chaitu236) [reply recipes-core/initrdscripts/files/ni_provisioning.safemode:325]: @billpittman can you create a story parented to hardknott work so we don't forget it.

**Review by chaitu236 — APPROVED** [review]

**Comment by billpittman** [comment]
Created a new US to force a safemode update and expanded the commit message a bit to include the impact of the change.

**Comment by billpittman** [comment]
Added code to remove the enable_initramfs variable from the grubenv to prevent older safemodes from attempting to use an initramfs to boot runmode.  As an intended side effect, the change to postinst also removes the previous requirement to force a safemode update.

**Review by amstewart — APPROVED** [review]
I assume the new content is in `nilrt-base-system-image.postinst`. If so, looks good to me.

**Review by shruthi-ravi — APPROVED** [review]

**Pull request merged**
